### DISCUSSION TITLE
Synchronization on QF for Android

### DISCUSSION THREAD

**Discussion by thymaro** [opengisch/QField#3212] [Q&A]
This question is mainly about mobile data volumes when syncing projects on mobile devices

**Process steps involved**

1. create a project on desktop, including 1GB COGs (to keep it simple)
2. deposit it on QFieldCloud (QFC)
3. download that project to several mobile devices (call them M1 and M2)
4. Record data in the field on M1 and push the changes
5. Record different data on M2 and push the changes
6. All edits are on the server now
7. Someone at the office synchronizes the project to desktop to see what the lads out in the rain are working on or whether they're just having a good time at the local bar. **Their sync data volume is only a few (kilo-/mega-)bytes**, because they don't need to download the COGs.
8. Synchronize the project on M1 to be able to see the edits by M2, and vice-versa on M2

**At step 8, how large is the transferred data? A few bytes as on desktop or are the 1GB of COGs downloaded again?** I am asking this because synching on QField for Android can take a long time and I have a suspicion that some large files are being downloaded again when this is not really necessary. Since QF for desktop QGIS is able to detect the changes and only download the necessary data, is QF for Android able to do the same or can/will this be implemented in the future? Knowing this will help us schedule the syncing tasks when out on a field day (i.e. every hour or so, or only during lunch, if the syncing speed cannot be improved).

**Comment by suricactus** [comment]
QField for Android does not download a file if it already exists and it has not been changed. 

I guess most of the time is lost on QFieldCloud, because everytime you project is prepared for Android usage, QFieldCloud needs to download the COG files internaly, which will take some time even within the same datacenter. You can check that on QFieldCloud Web interface, if you go to Your Project -> Jobs and check the latest `package` job start and end time.

Currently there is no way to speed this up.

**Reply by thymaro** [reply]
Thank you. That clears it up.
